packages/babel-traverse/src/path/introspection.ts

```diff
@@ -175,16 +175,18 @@ export function isStatementOrBlock(this: NodePath): boolean {
  */
 
 export function referencesImport(
-  this: NodePath<t.Expression>,
+  this: NodePath,
```

Comment:
Because `JSXMemberExpression` is not an `Expression`

Reply:
Sure, done
Though I wonder what's the point of narrowing the type of `this` here? all the other methods accept `NodePath` 🤔

Reply:
Uhm yes that's a good observation. This method always work, it just return `false` for other node types.

You are right, `NodePath` is good enough here. Sorry!

Reply:
Yeah, I had a thought that making `this` type narrower doesn't make it more convenient for the TypeScript user, since they either would already be in a context where the type matches (e.g in a visitor) and won't notice, or would be forced to preclude the call with a manual check, which is performed inside the method anyway.

> Sure, done

Reverted 😃

Reply:
Although...

```ts
function f(path: NodePath) {
  path.referencesImport("moduleSource", "importName");
}
```

this does not produce a TS error now 🤔

Reply:
Yeah that's correct, right? It can return `false` for every unsupported node type, but it always knows what to return.